deploy/terraform/modules/app_tier/nsg-webdisp.tf

```
@@ -1,31 +1,55 @@
+# Creates SAP web subnet nsg
+resource "azurerm_network_security_group" "nsg-web" {
+  count               = local.enable_deployment ? (local.sub_web_defined ? (local.sub_web_nsg_exists ? 0 : 1) : 0) : 0
```

Can be done the same way as `deploy/terraform/modules/app_tier/infrastructure.tf` to avoid complicated logic:
```suggestion
  count               = local.enable_deployment  && local.sub_web_defined ? (local.sub_web_nsg_exists ? 0 : 1) : 0
```